src/components/SearchFilter.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { Button } from '@blueprintjs/core';
+
+import queryString from 'query-string';
+import { debounce, isEqual, pickBy } from 'lodash';
+
+import SearchFilterEntities from './SearchFilterEntities';
+
+import './SearchFilter.css';
+
+class SearchFilter extends Component {
+  constructor(props)  {
+    super(props);
+
+    this.state = {
+      params: queryString.parse(props.location.search)
+    };
+
+    this.onTextChange = this.onTextChange.bind(this);
+    this.onEntityChange = this.onEntityChange.bind(this);
+
+    this.debouncedUpdate = debounce(this.updateLocation, 250);
+  }
+
+  updateLocation() {
+    const { history, location } = this.props;
+    const nonEmptyParams = pickBy(this.state.params, v => !!v);
+
+    history.push({
+      pathname: location.pathname,
+      search: queryString.stringify(nonEmptyParams)
+    });
+  }
+
+  componentWillUpdate(nextProps, { params }) {
+    if (!isEqual(params, this.state.params)) {
+      this.debouncedUpdate();
+    }
+  }
+
+  handleParamChange(param, newValue) {
+    this.setState({
+      params: {
+        ...this.state.params,
+        [param]: newValue
+      }
+    });
+  }
+
+  onTextChange(e) {
+    this.handleParamChange('q', e.target.value);
+  }
+
+  onEntityChange(type) {
```

Comment:
Entities are real things in Aleph, and this doesn't touch them. I would call it ``onSchemaChange``?